src/app/components/SearchContainer/SearchContainer.jsx

```diff
@@ -30,39 +30,49 @@ class SearchContainer extends React.Component {
   }
 
   componentDidUpdate(prevProps) {
-    if (this.props.location !== prevProps.location) {
+    const { location } = this.props;
+    if (location !== prevProps.location) {
       global.window.scrollTo(0, 0);
       this.loadSearch();
     }
   }
 
   loadSearch() {
-    const { query } = this.props.location;
-    const searchQuery = query && query.q ? query.q : this.props.searchQuery;
-    const selectedField = query && query.field ? query.field : this.props.searchField;
+    let { searchQuery } = this.props;
+    const {
+      location: { query },
+      dispatch,
+      searchField,
+    } = this.props;
+    searchQuery = query && query.q ? query.q : searchQuery;
+    const selectedField = query && query.field ? query.field : searchField;
     if (searchQuery) {
-      this.props.dispatch(searchActions.userQuery(searchQuery));
-      this.props.dispatch(searchActions.selectedField(selectedField));
-      this.props.dispatch(searchActions.searchPost(searchQuery, selectedField));
+      dispatch(searchActions.userQuery(searchQuery));
+      dispatch(searchActions.selectedField(selectedField));
+      dispatch(searchActions.searchPost(searchQuery, selectedField));
     }
   }
 
   render() {
-    const { query } = this.props.location;
+    const {
+      location: { query },
+    } = this.props;
     let { searchQuery } = this.props;
+    const { searchField, searchResults, eReaderUrl } = this.props;
```

Comment:
Why not merge the two destructures together?

Reply:
Fixed. Btw this comes from a eslint rule prefer-destructuring. As I saw too many errors in previous version with this, I disabled the rule. We can put it again and change all components if you think it's better.